classes/form/section_select_form.php

```diff
@@ -100,47 +102,79 @@ public function definition() {
         $massactionrequest = $this->_customdata['request'];
-        $data = \block_massaction\massactionutils::extract_modules_from_json($massactionrequest);
+        $data = massactionutils::extract_modules_from_json($massactionrequest);
         $modules = $data->modulerecords;
         $srcmaxsectionnum = max(array_map(function($mod) use ($sourcecoursemodinfo) {
             return $sourcecoursemodinfo->get_cm($mod->id)->sectionnum;
         }, $modules));
 
         $radioarray = [];
+
+        // Attributes for radio buttons.
+        $attributes = ['class' => 'mt-2'];
+
         // If user can add sections in target course or don't need to be able to.
-        if ($canaddsection || $srcmaxsectionnum <= $targetsectionnum) {
-            // We add the default value: Restore each course module to the section number it has in the source course.
-            $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
-            get_string('keepsectionnum', 'block_massaction'), -1, ['class' => 'mt-2']);
-        }
+        $cankeeporiginalsectionnum = ($canaddsection || $srcmaxsectionnum <= $targetsectionnum)
+            && massactionutils::can_keep_original_section_number($targetcourseid, $targetformat->get_format());
+        $this->is_allowed($cankeeporiginalsectionnum, -1, $attributes);
+        $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
+            get_string('keepsectionnum', 'block_massaction'), -1, $attributes);
 
         $sectionsrestricted = massactionutils::get_restricted_sections($targetcourseid, $targetformat->get_format());
         // Now add the sections of the target course.
         foreach ($targetsections as $sectionnum => $sectionname) {
-            $attributes = ['class' => 'mt-2'];
-            if (in_array($sectionnum, $sectionsrestricted)) {
-                $attributes['disabled'] = 'disabled';
-            }
+            $this->is_allowed(!in_array($sectionnum, $sectionsrestricted), $sectionnum, $attributes);
             $radioarray[] = $mform->createElement('radio', 'targetsectionnum',
                 '', $sectionname, $sectionnum, $attributes);
         }
 
-        if ($canaddsection) {
-            if (($targetsectionnum + 1) <= $targetformat->get_max_sections()) {
-                // New section option.
-                $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
-                    get_string('newsection', 'block_massaction'), $targetsectionnum + 1, ['class' => 'mt-2']);
-            }
-        }
+        // Whether user can add new section.
+        $canaddnewsection = $canaddsection && ($targetsectionnum + 1) <= $targetformat->get_max_sections();
+        $this->is_allowed($canaddnewsection, $targetsectionnum + 1, $attributes);
+        $radioarray[] = $mform->createElement('radio', 'targetsectionnum', '',
+            get_string('newsection', 'block_massaction'), $targetsectionnum + 1, $attributes);
+
```

Comment:
FOUND 1 ERROR AFFECTING 1 LINE
 140 | ERROR | [x] Functions must not contain multiple empty lines in a row; found 2 empty lines

Reply:
I have removed the redundant empty line

Reply:
Thanks @TomoTsuyuki for your review

I have updated the patch